google/cloud/storage/internal/openssl_util.cc

```diff
@@ -133,6 +133,56 @@ std::string OpenSslUtils::Base64Encode(std::string const& str) {
 #endif  // OPENSSL_IS_BORINGSSL
 }
 
+std::string OpenSslUtils::Base64Encode(std::vector<std::uint8_t> const& bytes) {
+#ifdef OPENSSL_IS_BORINGSSL
+  std::size_t encoded_size;
+  EVP_EncodedLength(&encoded_size, bytes.size());
+  std::vector<std::uint8_t> result(encoded_size);
+  std::size_t out_size =
+      EVP_EncodeBlock(result.data(), bytes.data(), bytes.size());
+  result.resize(out_size);
+  return {result.begin(), result.end()};
+#else
+  auto bio_chain = MakeBioChainForBase64Transcoding();
+  int retval = 0;
+
+  while (true) {
+    retval = BIO_write(static_cast<BIO*>(bio_chain.get()), bytes.data(),
+                       static_cast<int>(bytes.size()));
+    if (retval > 0) {
+      break;  // Positive value == successful write.
+    }
+    if (!BIO_should_retry(static_cast<BIO*>(bio_chain.get()))) {
+      std::ostringstream err_builder;
+      err_builder << "Permanent error in " << __func__ << ": "
+                  << "BIO_write returned non-retryable value of " << retval;
+      google::cloud::internal::ThrowRuntimeError(err_builder.str());
```

Comment:
Q. Presumably this means it is a "cannot happen" case?

Reply:
Yes, but I will probably fix more of this stuff in a separate cleanup PR.  Right now just trying to get the existing code to work for V4 signatures.